src/pkg/util/nvidiautils/nvidiautils.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package nvidiautils
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// GetNvidiaBindPath returns a string array consisting of filepaths of nvidia
+// related files to be added to the BindPaths
+func GetNvidiaBindPath(abspath string) []string {
+	var strArray []string
+	var bindArray []string
+	var searchArray []string
+	var commentID = regexp.MustCompile(`#`)
+	var soID = regexp.MustCompile(".so")
+
+	// use nvidia-container-cli (if present)
+	command, err := exec.LookPath("nvidia-container-cli")
+	if err == nil {
+		cmd := exec.Command(command, "list", "--binaries", "--ipcs", "--libraries")
+		out, err := cmd.Output()
+		if err == nil {
```

Comment:
```
if err != nil {
    return nil, fmt.Errorf("error running nvidia-container-cli: %v", err)
}
```

Reply:
Same comment as above.  As much as I dislike it, I think all of this code actually does need to be wrapped in the if statement b/c we don't want to bail if it doesn't work.

Reply:
Hmmm... nvidia-container-cli is an optional component. Don't want to return if error.

Reply:
Definitely want to error if we're unable to execute `nvidia-container-cli` if it does exist in `$PATH`

Reply:
No I disagree. `nvidia-container-cli` is still in a pretty early state afaik.  They have already changed he cli on me once.  If the user has a newer version of the binary installed that doesn't do what we expect we still need the option of falling back to the conf file instead.